source/apiVolontaria/volunteer/tests/tests_view_CyclesId.py

```diff
@@ -0,0 +1,181 @@
+from rest_framework.test import APIClient
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from volunteer.models import Cycle
+from apiVolontaria.factories import UserFactory, AdminFactory
+from django.utils import timezone
+import json
+
+
+class CyclesIdTests(APITestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+        self.user = UserFactory()
+        self.user.set_password('Test123!')
+        self.user.save()
+
+        self.admin = AdminFactory()
+        self.admin.set_password('Test123!')
+        self.admin.save()
+
+        self.cycle = Cycle.objects.create(
+            name='Cycle 1',
+            start_date=timezone.now(),
+            end_date=timezone.now() + timezone.timedelta(
+                minutes=100,
+            ),
+        )
+
+    def test_retrieve_cycle_id_not_exist(self):
+        """
+        Ensure we can't retrieve a cycle who doesn't exist.
+        """
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.get(
+            reverse(
+                'volunteer:cycles_id',
+                kwargs={'pk': 999},
+            )
+        )
+
+        content = {"detail": "Not found."}
+        self.assertEqual(json.loads(response.content), content)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_retrieve_cycle(self):
+        """
+        Ensure we can retrieve a cycle.
+        """
+        start_date = self.cycle.start_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        end_date = self.cycle.end_date.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+        data = {
+            "id": self.cycle.id,
+            "name": self.cycle.name,
+            "start_date": start_date,
+            "end_date": end_date,
+        }
+
+        self.client.force_authenticate(user=self.user)
+
+        response = self.client.get(
+            reverse(
+                'volunteer:cycles_id',
+                kwargs={'pk': self.cycle.id},
+            )
+        )
+
+        self.assertEqual(json.loads(response.content), data)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_update_cycle_with_permission(self):
```

Comment:
What happens if the cycle doesn't exist? (there is no test for that case)

Reply:
Good one, i will add it